sources/datasources/stories.js

```diff
@@ -0,0 +1,80 @@
+/** @format */
+
+const Events = require('../models/event.js');
+const Stories = require('../models/story.js');
+const currentStories = require("../models/currentStory");
+const Users = require('../models/user.js');
+const Clubs = require('../models/club.js');
+const Venues = require('../models/venue.js');
+const AccessLevel = require('../models/accessLevel.js');
+const { DataSource } = require('apollo-datasource');
+const { ApolloError } = require('apollo-server');
+
+class StoryAPI extends DataSource {
+	constructor() {
+		super();
+	}
+    initialize(config) {}
+    getStories(args) {
+		return Stories.find(args);
+	}
+	async getCurrentStories() {
+        let activeStories=await currentStories.find();
+        return activeStories;        
+	}
+	async addStory(story) {
+		let retPromise = {};
+		// Create Event with basic types;
+		let createdStory = await Stories.create({
+			asset: story.asset,
+            description: story.description,
+            isExpired:false
+		});
+
+		//Add nested types
+
+		//1. author
+		if (story.author != undefined) {
+			const authorId = story.author;
+            const foundAuthor = await Clubs.findById(authorId);
+            if(foundAuthor==undefined){
+                return new ApolloError("Author Not Found");
+            }
+            createdStory.author = foundAuthor._id;
+
+            //add to current Stories
+            await currentStories.create({
+                storyAsset:story.asset,                
+                authorLogo: foundAuthor.logo,
+                authorName: foundAuthor.clubName,
+                storyID: createdStory._id
+            })  
```

Comment:
Do we need an asset type in the model?
How to know if it is image or video down the line without trying to render the asset

Reply:
asset will be a cloudinary link

Reply:
yeah, but how will client know from link if it is video or image

Reply:
@Thesmader is working on that.